perfeed/data_stores/base.py

```diff
@@ -0,0 +1,32 @@
+from abc import ABC, abstractmethod
+from typing import Dict
+import pandas as pd
+from perfeed.models.pr_summary import PRSummary
+
+
+class UnsupportedFormatError(Exception):
+    """Exception raised for unsupported storage formats."""
+    pass
+
+
+class BaseStorage(ABC):
+    """Abstract base class for storage handlers."""
+    
+    def __init__(self, data_type: str, append: bool = True, overwrite: bool = False):
+        self.data_type = data_type
+        self.overwrite = overwrite
+        self.append = append
+        self._validate_options()
+
+    def _validate_options(self):
+        if self.overwrite and self.append:
+            raise ValueError("Cannot set both overwrite and append to True simultaneously.")
+
+    @abstractmethod
+    def save(self, data: PRSummary, metadata: Dict) -> None:
+        pass
+    
+    @abstractmethod
+    def load(self) -> pd.DataFrame:
```

Comment:
Is it common to return `pd.DataFrame` as a common type for different implementations?

Reply:
hard to common or not. the use case of `load` is more on the backend atm where we load to process info for weekly talk point or in the future, we load up for RAG purpose. Both scenario dataframe works well for speed (coz the volume we will get) and data processing.

Reply:
I see, but we already have a return type `PRSummary` in this case. Why don't we use it instead? a generic `pd.DataFrame` doesn't tell us what we get.

Reply:
wew likely will have many other types in the future and how do u plan to handle those? for the same reason, i use abstract type (BaseModel) for PRSummary and PRSUmmaryMetadata in `save` and `validate_and_convert` to keep input output type a bit general. dataframe is an equivalent of BaseModel. 
```
def save(self, data: BaseModel, metadata: BaseModel) -> None:
```

Reply:
Yeah, I think we can get away with a generic return type for now. To your question, as we have more types, we'll likely have to write separate `save` and `load` functions for them (especially if some are not a simple, direct mapping of the table, but require join or additional data to fill in). 

I'm not sure what's the idiomatic way to do that in python, but having multiple database getters for different types is pretty common in Java.

Reply:
the scope we have now is for PRsummary and TalkPoint. we can come back for this once we expand